Mathlib/LinearAlgebra/QuadraticForm/Basic.lean

```diff
@@ -820,65 +827,105 @@ open LinearMap (BilinMap)
 
 section AssociatedHom
 
-variable [CommRing R] [AddCommGroup M] [Module R M] [AddCommGroup N] [Module R N]
-variable (S) [CommSemiring S] [Algebra S R]
-variable [Module S N] [IsScalarTower S R N]
-variable [Invertible (2 : R)] {B₁ : BilinMap R M R}
+variable [CommRing R] [AddCommGroup M] [Module R M]
+
+/-- If `2` is invertible in `R`, then it is also invertible in `End R M`. -/
+instance [inst : Invertible (2 : R)] : Invertible (2 : Module.End R M) :=
+  (Invertible.map (algebraMap R (Module.End R M)) 2).copy 2 <|
+    (map_ofNat (algebraMap R (Module.End R M)) 2).symm
```

Comment:
This holds more generally as
```lean
variable [Semiring R] [AddCommMonoid M] [Module R M]

instance : SMulCommClass R (↥(Submonoid.center R)) M where
  smul_comm r r' m := by
    simp_rw [Submonoid.smul_def, smul_smul, (Set.mem_center_iff.1 r'.prop).1]

/-- If `2` is invertible in `R`, then it is also invertible in `End R M`. -/
instance [Invertible (2 : R)] : Invertible (2 : Module.End R M) where
  invOf :=  (⟨⅟2, Set.invOf_mem_center (Set.ofNat_mem_center _ _)⟩ : Submonoid.center R) • (1 : Module.End R M)
  invOf_mul_self := by
    ext m
    dsimp [Submonoid.smul_def]
    rw [← ofNat_smul_eq_nsmul R, invOf_smul_smul (2 : R) m]
  mul_invOf_self := by
    ext m
    dsimp [Submonoid.smul_def]
    rw [← ofNat_smul_eq_nsmul R, smul_invOf_smul (2 : R) m]
```
(which also preserves the code I suggested you delete elsewhere)